docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -837,15 +834,495 @@ spec:
+
+Examples of these plugins include {logstash-ref}/plugins-inputs-jdbc.html[`logstash-input-jdbc`] (which has no automatic way to split queries across {ls} instances), {logstash-ref}/plugins-inputs-s3.html[`logstash-input-s3`] (which has no way to split which buckets to read across {ls} instances), or {logstash-ref}/plugins-inputs-file.html[`logstash-input-file`].
+
+[id="{p}-logstash-inputs-external-state"]
+==== Input plugins: external source stores state
+
+{ls} installations that use input plugins that retrieve data from an external source, and **rely on the external source to store state** can scale based on the parameters of the external source.
+
+For example, a {ls} installation that uses a {logstash-ref}/plugins-inputs-kafka.html[`logstash-input-kafka`] plugin to retrieve data can scale the number of pods up to the number of partitions used, as a partition can have at most one consumer belonging to the same consumer group. 
+Any pods created beyond that threshold cannot be scheduled to receive data.
+
+Examples of these plugins include {logstash-ref}/plugins-inputs-kafka.html[`logstash-input-kafka`], {logstash-ref}/plugins-inputs-azure_event_hubs.html[`logstash-input-azure_event_hubs`], and {logstash-ref}/plugins-inputs-kinesis.html[`logstash-input-kinesis`].
+
+[id="{p}-logstash-working-with-plugin-considerations"]
+=== Plugin-specific considerations
+
+Some plugins have additional requirements and guidelines for optimal performance in a {ls} ECK environment.
+
+* <<{p}-logstash-plugin-considerations-ls-integration>>
+* <<{p}-logstash-plugin-considerations-es-output>>
+* <<{p}-logstash-plugin-considerations-integration-filter>>
+* <<{p}-logstash-plugin-considerations-agent-beats>>
+
+TIP: Use these guidelines _in addition_ to the general guidelines provided in <<{p}-logstash-working-with-plugins-scaling>>.
+
+[id="{p}-logstash-plugin-considerations-ls-integration"]
+==== {ls} integration plugin 
+
+When your pipeline uses the {logstash-ref}/plugins-integrations-logstash.html[`Logstash integration`] plugin, add `keepalive=>false` to the {logstash-ref}/plugins-outputs-logstash.html[logstash-output] definition to ensure that load balancing works correctly rather than keeping affinity to the same pod.
+
+[id="{p}-logstash-plugin-considerations-es-output"]
+==== Elasticsearch output plugin
+
+The {logstash-ref}/plugins-output-elasticsearch.html[`elasticsearch output`] plugin requires certain roles to be configured in order to enable {ls} to communicate with {es}.
+
+[source, logstash]
+----
+"cluster": ["monitor", "manage_ilm", "read_ilm", "manage_logstash_pipelines", "manage_index_templates", "cluster:admin/ingest/pipeline/get",],
+  "indices": [
+    {
+      "names": [ "logstash", "logstash-*", "ecs-logstash", "ecs-logstash-*", "logs-*", "metrics-*", "synthetics-*", "traces-*" ],
+      "privileges": ["manage", "write", "create_index", "read", "view_index_metadata"]
+    }
+----
```

Comment:
```suggestion
You can customize roles in {es}. Check out <<{p}-users-and-roles,creating custom roles>>

[source, logstash]
----
kind: Secret
apiVersion: v1
metadata:
  name: my-roles-secret
stringData:
  roles.yml: |-
    eck_logstash_user_role:
      "cluster": ["monitor", "manage_ilm", "read_ilm", "manage_logstash_pipelines", "manage_index_templates", "cluster:admin/ingest/pipeline/get",],
      "indices": [
        {
          "names": [ "logstash", "logstash-*", "ecs-logstash", "ecs-logstash-*", "logs-*", "metrics-*", "synthetics-*", "traces-*" ],
          "privileges": ["manage", "write", "create_index", "read", "view_index_metadata"]
        }
      ]
----
```

Reply:
I added a pointer to https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-users-and-roles.html#k8s_creating_custom_roles which has the necessary steps to customzie roles. 
It is not a simple step just k apply -f resources.yaml. It requires configuration in ES ECK

Reply:
^^ I'm not seeing the pointer. Where did you add it?